### PR TITLE
Fix Resource.meta.lastUpdated property not being updated during inserts and updates

### LINKED ISSUE #2391: [P2P] lastUpdated not updated for some records
- [ ] The `Resource.meta.lastUpdated` is not updated for resources downloaded from the server
- [ ] The `Resource.meta.lastUpdated` is not updated for generated tasks
- [ ] The `Resource.meta.lastUpdated` might not be updated for some resources. Generally, `FhirEngine.add` or `FhirEngine.update` should not be used and `DefaultRepository.addOrUpdate` should be used since it updates the `Resource.meta.lastUpdated`

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #2391

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/di/FhirEngineModule.kt [hunk 1]
```diff
@@ -54,7 +53,7 @@ class FhirEngineModule {
   ): FhirEngine {
     FhirEngineProvider.init(
       FhirEngineConfiguration(
-        enableEncryptionIfSupported = !BuildConfig.DEBUG,
+        enableEncryptionIfSupported = false,
```
[the remark this answers is not in the document]
Comment (Rkareko) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/di/FhirEngineModule.kt:56]: Reminder to act on the `TODO`

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderView.kt [hunk 2]
```diff
@@ -58,7 +58,7 @@ fun LoaderDialog(
   val openDialog = remember { mutableStateOf(true) }
   if (openDialog.value) {
     Dialog(
-      onDismissRequest = { openDialog.value = true },
+      onDismissRequest = { openDialog.value = false },
```
Comment (Rkareko) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderView.kt:61]: What issue is fixed by this?

Comment (ekigamba) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderView.kt:61]: Nice catch. I use this during testing only

**Review by ellykits — COMMENTED** [review]
@ekigamba LGTM :+1: . Just revert the lines you had updated for testing and we'll be ready to go. @Rkareko already left some comments.

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/FhirCoreDownloadManager.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import com.google.android.fhir.sync.Request
+import com.google.android.fhir.sync.download.ResourceParamsBasedDownloadWorkManager
+import com.google.android.fhir.sync.download.ResourceSearchParams
+import org.hl7.fhir.r4.model.Resource
+import org.hl7.fhir.r4.model.ResourceType
+import org.smartregister.fhircore.engine.data.local.DefaultRepository
+import org.smartregister.fhircore.engine.util.extension.updateLastUpdated
+
+/** Created by Ephraim Kigamba - nek.eam@gmail.com on 13-06-2023. */
+class FhirCoreDownloadManager(
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/sync/FhirCoreDownloadManager.kt:29]:
```suggestion
class OpenSrpDownloadManager(
```

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/sync/FhirCoreDownloadManager.kt:29]: Can we rename this to OpenSrp in line with the upcoming package name refactor?

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**